go.mod

```diff
@@ -1,33 +1,98 @@
 module github.com/vmware-tanzu/load-balancer-operator-for-kubernetes
 
-go 1.16
+go 1.17
 
 require (
 	github.com/bitly/go-simplejson v0.5.0
-	github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869 // indirect
-	github.com/go-logr/logr v0.4.0
+	github.com/go-logr/logr v1.2.2
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/onsi/ginkgo v1.16.4
-	github.com/onsi/gomega v1.16.0
+	github.com/onsi/ginkgo v1.16.5
+	github.com/onsi/gomega v1.18.1
 	github.com/pkg/errors v0.9.1
 	github.com/satori/go.uuid v1.2.0
+	github.com/vmware-tanzu/carvel-kapp-controller v0.35.0
+	github.com/vmware-tanzu/tanzu-framework v0.24.0
```

Comment:
t-f repo is a huge one, is it possible that we just include a part of this? I believe @chenlin07  has some experience here, you may want to talk to her on this